Automattic/simplenote-android
# Add WP.com Sign In

This PR adds support for signing to Simplenote via your WordPress.com account. It supports creating a new Simplenote account if one didn't exist yet, or alternatively it will sign the user to an existing account.

I utilized this library to handle the oauth work: https://github.com/openid/AppAuth-Android

![wpcom-sign-in](https://user-images.githubusercontent.com/789137/40395993-92f06cac-5ddf-11e8-9c1b-23910530e40b.gif)

**To Test**
* From the login screen tap 'Sign in with WordPress.com'
* Authorize a WordPress.com account
* 3 things should happen:
 ** If a Simplenote account already exists for the account, it should be signed in successfully.
 ** If a Simplenote account didn't exist for the account, a new one should be created and you'll see the welcome note.
 ** If the WordPress.com account hasn't been verified by email, you should get an error message.

We're storing the WP.com token in the keychain for future integrations. If a user signed out, it will be deleted.

_Note: The reviewer will need to add the client id to the config file. Ping me on slack to get the ID!_
